src/Faker/Provider/Base.php

```diff
@@ -168,9 +168,17 @@ public static function randomAscii()
      *
      * @return array New array with $count elements from $array
      */
-    public static function randomElements(array $array = array('a', 'b', 'c'), $count = 1, $allowDuplicates = false)
+    public static function randomElements($array = array('a', 'b', 'c'), $count = 1, $allowDuplicates = false)
     {
-        $allKeys = array_keys($array);
+        if ($array instanceof \Traversable) {
+            foreach ($array as $element) {
+                $traversables[] = $element;
```

Comment:
You need to set `traversables` to an empty array first I guess.

Reply:
i could, but this is why i use the `isset` below.  it's 2 ways to accomplish the same thing.

Reply:
Then use only 1 intermediate variable if you want, but pushing to a non-existing array will cause a warning. If it doesn't, it's a bad practice anyway.